Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py

```diff
@@ -0,0 +1,448 @@
+        "destination": args.get('destination'),
+        "tag": args.get('tag'),
+        "limit": args.get('limit'),
+        "incident_type": args.get('incident_type')
+    }
+    demisto.debug(f'GuardiCoreV2 - Fetch incidents parameters: {fetch_params}')
+    results = client.get_incidents(fetch_params)
+    demisto.debug(
+        f'GuardiCoreV2 - Fetch incidents results count: {len(results)}')
+
+    incidents = []
+    for inc in results.get('objects'):
+        id = inc.get('_id')
+        start_time = inc.get('start_time')
+        if not id or "-" not in id or not start_time:
+            demisto.debug(
+                'GuardiCoreV2 - Fetch incidents: skipped fetched incident because no start time or id')
+            continue
+
+        incident = {
+            'name': f"INC-{id.split('-')[0].upper()}",
+            'occurred': timestamp_to_datestring(start_time, DATE_FORMAT),
+            'rawJSON': json.dumps(inc)
+        }
+        incidents.append(incident)
+
+        if current_fetch < start_time:
+            current_fetch = start_time
+
+    demisto.debug(
+        f'GuardiCoreV2 - Fetch incidents: fetch time finished at: {current_fetch}')
+    return incidents, current_fetch
+
+
+def get_indicent(client: Client, args: Dict[str, Any]) -> CommandResults:
+    incident_id = args.get('id', None)
+    if not incident_id:
+        raise DemistoException(
+            "GuardiCoreV2 - get incident needs an id parameter.")
+
+    # Call the Client function and get the raw response
+    result = client.get_incident(incident_id)
+
+    hr = filter_human_readable(result, human_columns=INCIDENT_COLUMNS)
+    md = tableToMarkdown(f'GuardiCoreV2 - Incident: {incident_id}', hr)
+
+    return CommandResults(
+        outputs_prefix='Guardicore.Incident',
+        outputs_key_field=incident_id,
+        readable_output=md,
+        outputs=result,
+    )
+
+
+def get_assets(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
+    ip_address = args.get('ip_address', None)
+    name = args.get('name', None)
+    asset_id = args.get('asset_id', None)
+    limit = int(args.get('limit', 50))
+    offset = int(args.get('offset', 0))
+
+    if not ip_address and not name and not asset_id:
+        raise DemistoException(
+            "GuardiCoreV2 - Endpoint search must have ip, name or asset_id defined.")
+
+    params = {
+        "asset_id": asset_id,
+        "limit": limit,
+        "offset": offset
+    }
+
+    if ip_address:
+        params['search'] = ip_address
+    elif name:
+        params['search'] = name
+
+    results = client.get_assets(params)
+
+    endpoints = []
+    for res in results.get("objects"):
+        hostname = res.get("guest_agent_details", {}).get("hostname", "")
+
+        res = filter_human_readable(res, human_columns=ASSET_COLUMNS)
+        md = tableToMarkdown(f'GuardiCoreV2 - Asset: {hostname}',
+                             res)
+
+        endpoints.append(CommandResults(
```

Comment:
we also usually have a key to identify on. meaning in your case, that the assets in the context will be merged if the asset has the same id.
see: https://github.com/demisto/content/blob/2b388d5f8d074485ac7c1a592a1a45374d8f7971/Packs/Zimperium/Integrations/Zimperium/Zimperium.py#L601